engine/src/test/java/io/zeebe/engine/processor/workflow/job/ActivatableJobsNotificationTests.java

```diff
@@ -0,0 +1,149 @@
+      (type) ->
+          Bpmn.createExecutableProcess(PROCESS_ID)
+              .startEvent("start")
+              .serviceTask("task", b -> b.zeebeTaskType(type).done())
+              .endEvent("end")
+              .done();
+
+  private String taskType;
+
+  private static Consumer<String> jobAvailableCallback =
+      (Consumer<String>) Mockito.spy(Consumer.class);
+
+  @ClassRule
+  public static final EngineRule ENGINE =
+      EngineRule.singlePartition().withJobsAvailableCallback(jobAvailableCallback);
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Before
+  public void setup() {
+    taskType = Strings.newRandomValidBpmnId();
+    ENGINE.deployment().withXmlResource(PROCESS_ID, MODEL_SUPPLIER.apply(taskType)).deploy();
+  }
+
+  @Test
+  public void shouldNotifyWhenFirstJobCreated() {
+    // when
+    createWorkflowInstanceAndJobs(1);
+
+    // then
+    Mockito.verify(jobAvailableCallback, times(1)).accept(taskType);
+  }
+
+  @Test
+  public void shouldNotNotifyWhenSecondJobCreated() {
+    // when
+    createWorkflowInstanceAndJobs(2);
+
+    // then
+    Mockito.verify(jobAvailableCallback, times(1)).accept(taskType);
+  }
+
+  @Test
+  public void shouldNotifyWhenJobsAvailableAgain() {
+    // given
+    createWorkflowInstanceAndJobs(2);
+    final Record<JobBatchRecordValue> jobs = activateJobs(2);
+
+    // when
+    createWorkflowInstanceAndJobs(1);
+
+    // then
+    Mockito.verify(jobAvailableCallback, times(2)).accept(taskType);
+  }
+
+  @Test
+  public void shouldUpdateCountWhenJobCanceled() {
+    // given
+    final List<Long> instanceKeys = createWorkflowInstanceAndJobs(1);
+    ENGINE.workflowInstance().withInstanceKey(instanceKeys.get(0)).cancel();
+
+    // when
+    createWorkflowInstanceAndJobs(1);
+
+    // then
+    Mockito.verify(jobAvailableCallback, times(2)).accept(taskType);
+  }
+
+  @Test
+  public void shouldNotUpdateCountWhenActivatedJobCanceled() {
+    // given
+    final List<Long> instanceKeys = createWorkflowInstanceAndJobs(2);
+    activateJobs(1);
+    ENGINE.workflowInstance().withInstanceKey(instanceKeys.get(0)).cancel();
+
+    // when
+    createWorkflowInstanceAndJobs(1);
+
+    // then
+    Mockito.verify(jobAvailableCallback, times(1)).accept(taskType);
+  }
+
+  private List<Long> createWorkflowInstanceAndJobs(int amount) {
+    final List<Long> instanceKeys =
+        IntStream.range(0, amount)
+            .boxed()
+            .map(i -> ENGINE.workflowInstance().ofBpmnProcessId(PROCESS_ID).create())
+            .collect(Collectors.toList());
+    assertThat(
+            jobRecords(JobIntent.CREATED)
+                .withType(taskType)
+                .filter(r -> instanceKeys.contains(r.getValue().getWorkflowInstanceKey()))
+                .limit(amount)
+                .count())
+        .isEqualTo(amount);
+    return instanceKeys;
```

Comment:
```suggestion
    return IntStream.range(0, amount)
      .mapToObj(i -> ENGINE.createJob(taskType, PROCESS_ID))
      .map(r -> r.getValue().getWorkflowInstanceKey())
      .collect(Collectors.toList());
```

We can use our existing logic here.